File: CS/MapPieItem/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace MapPieItem {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // Create a layer to show vector items.
            VectorItemsLayer itemsLayer = new VectorItemsLayer();
            mapControl1.Layers.Add(itemsLayer);

            // Create a storage to provide data for the vector layer.
            MapItemStorage storage = new MapItemStorage();
            itemsLayer.Data = storage;

            // Create a pie with several segments.
            MapPie pie = new MapPie();
            pie.Size = 200;
            pie.Segments.Add(new PieSegment() { Argument = "A", Value = 100 });
            pie.Segments.Add(new PieSegment() { Argument = "B", Value = 50 });
            pie.Segments.Add(new PieSegment() { Argument = "C", Value = 120 });
            storage.Items.Add(pie);

            // Provide color indexes to bubbles as attributes.
            int i = 0;
            foreach (PieSegment segment in pie.Segments) {
                segment.Attributes.Add(new MapItemAttribute() { Name = "Color", Value = i });
                i++;
            }

            // Create a colorizer to provide colors for bubble items.
            ColorIndexColorizer colorizer = new ColorIndexColorizer();
            itemsLayer.Colorizer = colorizer;

            // Add colors to the colorizer.
            colorizer.ColorItems.Add(new ColorizerColorTextItem() { Color = Color.Coral, Text = "Category A" });
            colorizer.ColorItems.Add(new ColorizerColorTextItem() { Color = Color.Orange, Text = "Category B" });
            colorizer.ColorItems.Add(new ColorizerColorTextItem() { Color = Color.LightBlue, Text = "Category C" });

            // Load color indexes from bubbles via the 'Color' attribute
            colorizer.ColorIndexProvider = new PieSegmentAttributeToColorIndexProvider() { AttributeName = "Color" };


            // Show a color legend.
            mapControl1.Legends.Add(new ColorListLegend() { Layer = itemsLayer });

        }
    }
}

File: VB/MapPieItem/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace MapPieItem
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' Create a layer to show vector items.
			Dim itemsLayer As New VectorItemsLayer()
			mapControl1.Layers.Add(itemsLayer)

			' Create a storage to provide data for the vector layer.
			Dim storage As New MapItemStorage()
			itemsLayer.Data = storage

			' Create a pie with several segments.
			Dim pie As New MapPie()
			pie.Size = 200
			pie.Segments.Add(New PieSegment() With {.Argument = "A", .Value = 100})
			pie.Segments.Add(New PieSegment() With {.Argument = "B", .Value = 50})
			pie.Segments.Add(New PieSegment() With {.Argument = "C", .Value = 120})
			storage.Items.Add(pie)

			' Provide color indexes to bubbles as attributes.
			Dim i As Integer = 0
			For Each segment As PieSegment In pie.Segments
				segment.Attributes.Add(New MapItemAttribute() With {.Name = "Color", .Value = i})
				i += 1
			Next segment

			' Create a colorizer to provide colors for bubble items.
			Dim colorizer As New ColorIndexColorizer()
			itemsLayer.Colorizer = colorizer

			' Add colors to the colorizer.
			colorizer.ColorItems.Add(New ColorizerColorTextItem() With {.Color = Color.Coral, .Text = "Category A"})
			colorizer.ColorItems.Add(New ColorizerColorTextItem() With {.Color = Color.Orange, .Text = "Category B"})
			colorizer.ColorItems.Add(New ColorizerColorTextItem() With {.Color = Color.LightBlue, .Text = "Category C"})

			' Load color indexes from bubbles via the 'Color' attribute
			colorizer.ColorIndexProvider = New PieSegmentAttributeToColorIndexProvider() With {.AttributeName = "Color"}


			' Show a color legend.
			mapControl1.Legends.Add(New ColorListLegend() With {.Layer = itemsLayer})

		End Sub
	End Class
End Namespace
